apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveDao.kt

```diff
@@ -216,18 +219,22 @@ class OppgaveDaoImpl(
             }
         }
 
-    override fun hentOppgaverForSak(sakId: SakId): List<OppgaveIntern> =
+    override fun hentOppgaverForSak(
+        sakId: SakId,
+        type: OppgaveType?,
+    ): List<OppgaveIntern> =
         connectionAutoclosing.hentConnection {
             with(it) {
                 val statement =
                     prepareStatement(
                         """
                         SELECT o.id, o.status, o.enhet, o.sak_id, o.type, o.saksbehandler, o.referanse, o.merknad, o.opprettet, o.saktype, o.fnr, o.frist, o.kilde, o.forrige_saksbehandler, si.navn
                         FROM oppgave o LEFT JOIN saksbehandler_info si ON o.saksbehandler = si.id
-                        WHERE o.sak_id = ?
+                        WHERE o.sak_id = ? ${type?.name?.let { " and o.type = ?" } ?: ""}
```

Comment:
Veit ikkje om vi har noko betre mønster for dette? 🤔

Reply:
Eventuelt iterere over oppgavetype-enumen, og køyre det som ein where in 🤔

Reply:
Kan skrive 
`(? OR o.type = ?)` 
...med...
```
statement.setBoolean(2, type == null)
statement.setString(3, type?.name)
``` 

Ikke fryktelig mye enklere da, men gjort en par andre plasser i koden vår... 🤷‍♂️

Reply:
Vi slipper det om vi splitter det opp i to db-metoder også 🤔

Reply:
Enda bedre ☝️ 😄

Reply:
Prøver ein variant med to service-metodar, synes denne vart ganske så ålreit.